tests/integration_tests/test_ray.py

```diff
@@ -0,0 +1,134 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2019 Uber Technologies, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import os
+import shutil
+import tempfile
+
+import dask.dataframe as dd
+import pytest
+import ray
+import tensorflow as tf
+
+from ludwig.api import LudwigModel
+from ludwig.backend.ray import RayBackend, get_horovod_kwargs
+
+from tests.integration_tests.utils import create_data_set_to_use
+from tests.integration_tests.utils import bag_feature
+from tests.integration_tests.utils import binary_feature
+from tests.integration_tests.utils import category_feature
+from tests.integration_tests.utils import date_feature
+from tests.integration_tests.utils import generate_data
+from tests.integration_tests.utils import h3_feature
+from tests.integration_tests.utils import numerical_feature
+from tests.integration_tests.utils import sequence_feature
+from tests.integration_tests.utils import set_feature
+from tests.integration_tests.utils import spawn
+from tests.integration_tests.utils import text_feature
+from tests.integration_tests.utils import vector_feature
+
+
+@pytest.fixture
+def ray_start_4_cpus():
+    res = ray.init(num_cpus=4)
+    try:
+        yield res
+    finally:
+        ray.shutdown()
+
+
+def train_with_backend(backend, config, dataset=None, training_set=None, validation_set=None, test_set=None):
```

Comment:
if it's the same as the one in the dask test file, maybe we could put them in the util file?

Reply:
Good catch. Fixed.